### PR TITLE
Detect Spring Boot fat-JAR repackaging and use original thin JAR

### PR DESCRIPTION
**Pull request by chanseokoh**
Implements https://github.com/GoogleContainerTools/jib/issues/1852#issuecomment-557275771 for Maven. That is, auto-detect Spring Boot fat-JAR repackaging and use the original thin JAR.

Additionally takes care of the corner cases with `<classifier>` and `<outputDirectory>` from `maven-jar-plugin`. This is basically irrelevant of the Spring Boot fat JAR stuff, but I decided to fix them for correctness.

### CODE REVIEW

File: jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/MavenProjectProperties.java [hunk 1]
```diff
@@ -427,9 +430,54 @@ public boolean isOffline() {
...
+      consoleLogger.log(
+          Level.LIFECYCLE, "Spring Boot repackaging (fat JAR) detected; using the original JAR");
+      if (outputDirectory.equals(buildDirectory)) { // Spring renames original only when needed
+        suffix += ".original";
+      }
+    }
+
+    String jarName =
+        project.getBuild().getFinalName() + (classifier == null ? "" : '-' + classifier) + suffix;
+    consoleLogger.log(Level.LIFECYCLE, "Using JAR: " + outputDirectory.resolve(jarName));
```
Comment (TadCordle) [inline jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/MavenProjectProperties.java]: Does this need to be logged at `lifecycle`? Would be nice to keep log output minimal.

**Comment by chanseokoh** [comment]
Oh, wait. Something's not right. Looking into the issue.

**Comment by chanseokoh** [comment]
Added a Spring Boot integration test that verifies the image is built with a thin JAR and prints out "Hello world".

While doing that, I realized that JVM does not interpret the wildcard (`*`) in `java -cp /some/path/*`. Unfortunately, it has to end with `.jar` for JAR files. However, Spring Boot renames the original JAR with the `.jar.original` suffix, so we need to make the JAR filename end with `.jar` inside the image. The simplest solution I think is to copy the original JAR to a temp location for the sake of having the `.jar` suffix.

**Comment by chanseokoh** [comment]
Okay, let's give this a spin. This won't make things worse even if this goes wrong, because the packaged mode has not been working for Spring Boot projects.

**Pull request merged**